content/library/advanced-features/app-testing.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Getting started with app testing
+slug: /library/advanced-features/app-testing
+---
+
+# Getting started with app testing
+
+Streamlit app testing enables developers to build and run headless tests that execute their app code directly, simulate user input, and inspect rendered outputs for correctness.
+
+The provided class, AppTest, simulates a running app and provides methods to set up, manipulate, and inspect the app contents via API instead of a browser UI. It can be used to write automated tests of an app in various scenarios. These can then be run using a tool like [pytest](https://docs.pytest.org/). A typical pattern is to build a suite of tests for an app to ensure consistent functionality as the app evolves. The tests run locally and/or in a CI environment like Github Actions.
+
+## Using pytest
```

Comment:
Jumping right into a tutorial. Might want to set context with something like: "This tutorial shows how to author and run tests for a Streamlit app using AppTest and pytest:

- Install pytest and set up your repository
- Write a simple test using AppTest
- Execute the test and view results"

Reply:
Now that I've written the rest of the guide, I've added a bit more foreshadowing. I don't intend this to be a tutorial per se, though there is a sort of mini-tutorial within it to just run a given example. Please let me know if this needs further edits. 😃

Reply:
Nice! LGTM